src/widget/city_without_overlay.c

```diff
@@ -335,32 +335,91 @@ static void draw_figures(int x, int y, int grid_offset)
     }
 }
 
-static void draw_dock_workers(const building *b, int x, int y, color_t color_mask)
+static int fumigation_direction = 1;
+
+static void draw_fumigation(building *b, int x, int y, color_t color_mask)
 {
-    int num_dockers = building_dock_count_idle_dockers(b);
-    if (num_dockers > 0) {
-        int image_dock = map_image_at(b->grid_offset);
-        int image_dockers = image_group(GROUP_BUILDING_DOCK_DOCKERS);
-        if (image_dock == image_group(GROUP_BUILDING_DOCK_1)) {
-            image_dockers += 0;
-        } else if (image_dock == image_group(GROUP_BUILDING_DOCK_2)) {
-            image_dockers += 3;
-        } else if (image_dock == image_group(GROUP_BUILDING_DOCK_3)) {
-            image_dockers += 6;
+    int image_id = 5354; // smoke image_id
+    image_id += b->fumigation_frame;
+    image_draw_masked(image_id, x, y, color_mask);
+    if (image_id == 5357) {
+        fumigation_direction = 0;
+    }
+    if (image_id == 5354) {
+        fumigation_direction = 1;
+    }
+    building_animation_advance_fumigation(b, fumigation_direction);
+}
+
+static void draw_plague(building *b, int x, int y, color_t color_mask)
```

Comment:
Beware of the `const`s here and below: here, it seems `building *b` can be `const building *b`, and the same in the functions below.

Reply:
I change b->fumigation_direction now, so it can't be const